packages/config/src/index.ts

```diff
@@ -40,10 +40,17 @@ class TruffleConfig {
     );
   }
 
-  public eventManagerOptions(config: TruffleConfig): any {
-    let muteLogging;
-    const { quiet, logger, subscribers } = config;
-    return { logger, quiet, subscribers };
+  public eventManagerOptions(options: any): any {
+    const { quiet, logger, subscribers } = options;
+    const eventManagerOptions: any = { quiet, logger, subscribers };
+    // filter undefined values
+    return Object.keys(eventManagerOptions)
```

Comment:
I see what this loop is doing, but the way it goes about it is fairly awkward.  It might be simpler to just mutate `eventManagerOptions`.  Like, for instance,

```javascript
for (let key in eventManagerOptions) {
  if (eventManagerOptions[key] === undefined) {
    delete eventManagerOptions[key];
  }
}
return eventManagerOptions;
```

I think something like that would be easier to read.  Also you don't need to use `typeof` just to check for undefined. :)

Reply:
Ok I'll go ahead and make this easier to read. I wanted to find a more functional way to do this.

Reply:
So I ended up using `_.pick` for this method because it more cleanly does exactly what I'm trying to do here :)

Reply:
Oh, wonderful!